src/components/Cards/EditAccountModal.js

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import { auth, db } from "../../firebaseConfig";
+import { doc, updateDoc } from "firebase/firestore";
+import { updateEmail, signOut  } from "firebase/auth";
+import ShelterFrom from './ShelterForm';
+import AdoptFrom from './AdoptForm';
+import { useNavigate } from 'react-router-dom';
+
+const EditAccountModal = ({ showModal, setShowModal, user, setUser, isShelter }) => {
+
+    const navigate = useNavigate();
+    const [currentUser, setCurrentUser] = useState(user);
+
+    const handleChange = (e) => {
+        setCurrentUser({...currentUser, [e.target.name]: e.target.value });
+    };
+
+    useEffect(() => {
+        setCurrentUser(user);
+    }, [user]);
+
+    const handleModalClose = () => {
+        setShowModal(false);
+    };
+
+    const handleUpdateAccount = async () => {
+
+        let curAccountBody = { 
+            ...currentUser
+        }
+
+        delete curAccountBody.id;
+
+        // If email changed update Firebase Auth
+        if (curAccountBody.email !== auth.currentUser.email) {
+
+            try {
+                await updateEmail(auth.currentUser, curAccountBody.email);
+            }
+            // Redirect back to login if authentication session expired
+            catch (e) {
+                if(e) {
+                    signOut(auth).then(() => {
+                        navigate("/login")
```

Comment:
I think that should go to `sign-in` instead of `login`

Reply:
Nice catch, I'm not sure how it worked when I was testing it.